internal/gensym/ulid.go

```diff
@@ -24,14 +24,14 @@ func NewULIDGenerator(ctx context.Context) *ULIDGenerator {
 	}
 }
 
-func (gen *ULIDGenerator) NextID(ctx context.Context) ([]byte, error) {
+func (gen *ULIDGenerator) NextID(ctx context.Context) (ulid.ULID, error) {
 	// The math/rand generator is not thread-safe, so we have to guard access with a mutex.
 	gen.mu.Lock()
 	defer gen.mu.Unlock()
 	ts := ulid.Timestamp(chrono.Now(ctx))
 	id, err := ulid.New(ts, gen.entropy)
```

Comment:
If not adding additional context to error why not just
```suggestion
	return ulid.New(ts, gen.entropy)
```

Reply:
Good catch. This was slightly different code before and I just didn't do the cleanup.